### PR TITLE
[OSD-12370] Fixing metrics and adding local testing changes

### PR DESCRIPTION
**Pull request by ravitri**
### What type of PR is this?
_(bug)_ 

### What this PR does / why we need it?

Based on [OSD-12370](https://issues.redhat.com/browse/OSD-12370), this PR is meant to fix the mismatch of labels between the pod, svc and servicemonitor. However, the namespace and subjectpermission controllers do not enable the metrics for RPO yet. At this point, there are no alerts configured. Regarding the metrics, there needs to be more thought given on what kind of alerts required and then tune the metrics around it. 

Additionally, there are other local development and maintenance fixes/changes which are part of the PR.

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_  [OSD-12370](https://issues.redhat.com/browse/OSD-12370)

### Special notes for your reviewer:
- Added some logic to handle "local" run of the operator as well since that was missing.

### Pre-checks (if applicable):
- [x] Tested latest changes against a cluster

### CODE REVIEW

**Comment by ravitri** [comment]
/hold
Keeping PR on hold to fix the PR tests

File: main.go [hunk 1]
```diff
@@ -120,11 +150,17 @@ func main() {
 		os.Exit(1)
 	}
 
-	metricsServer := osdmetrics.NewBuilder(config.OperatorNamespace, "localmetrics-"+config.OperatorName).
+	ns, err := k8sutil.GetOperatorNamespaceEnv()
```
Comment (mrbarge) [inline main.go:153]: Can we re-use the namespace from the earlier call on line 117?

Comment (ravitri) [reply main.go]: Thank you Matt! Fixed it

**Pull request merged**